packages/database/src/core/Repo.ts

```diff
@@ -458,13 +458,19 @@ export function repoGetValue(repo: Repo, query: QueryContext): Promise<Node> {
   }
   return repo.server_.get(query).then(
     payload => {
-      const node = nodeFromJSON(payload as string);
+      const node = nodeFromJSON(payload as string).withIndex(
+        query._queryParams.getIndex()
+      );
       const events = syncTreeApplyServerOverwrite(
         repo.serverSyncTree_,
         query._path,
         node
       );
-      eventQueueRaiseEventsAtPath(repo.eventQueue_, query._path, events);
+      let affectedPath = query._path;
+      if (events.length > 0) {
+        affectedPath = repoRerunTransactions(repo, query._path);
```

Comment:
I think we already call this:

https://github.com/firebase/firebase-js-sdk/blob/master/packages/database/src/core/PersistentConnection.ts#L204

which invokes
https://github.com/Firebase/firebase-js-sdk/blob/master/packages/database/src/core/Repo.ts#L396